### PR TITLE
Add PlanRadioButton Component

### PR DESCRIPTION
**Pull request by robertf4**
#### Changes proposed in this Pull Request:
* This adds a PlanRadioButton component that will be used in https://github.com/Automattic/jetpack/pull/14018.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
See the P2 here for the design of this PR: p1HpG7-7MK-p2 (specifically the part titled "WP Admin Desktop")
See the P2 here for the overall MT: p1HpG7-7ET-p2

#### Testing instructions:
This cannot be tested on its own here. Use https://github.com/Automattic/jetpack/pull/14018 to test.

#### Proposed changelog entry for your changes:
* Not needed

### CODE REVIEW

**Review by delawski — CHANGES_REQUESTED** [review]
@robertf4 Thank you for extracting this presentational component to a separate PR. I left some feedback regarding the CSS.

I know you've got a lot on your plate right now, so if you need any support in addressing the feedback, feel free to reach out to me.

File: _inc/client/plans/single-product-backup.scss [hunk 1]
```diff
@@ -0,0 +1,11 @@
+.plan-radio-button__input {
+	margin-left: -20px !important;
+
+	@media ( max-width: 782px ) {
```
Comment (delawski) [inline _inc/client/plans/single-product-backup.scss]: Any reason why we are not using the `breakpoint()` SCSS mixin here?

File: _inc/client/plans/single-product-backup.scss [hunk 2]
```diff
@@ -0,0 +1,11 @@
+.plan-radio-button__input {
+	margin-left: -20px !important;
```
[the remark this answers is not in the document]
Comment (robertf4) [reply _inc/client/plans/single-product-backup.scss]: Agreed, much cleaner. Updated.

File: _inc/client/plans/single-product-backup.scss (hunk 1 above)
Comment (robertf4) [reply _inc/client/plans/single-product-backup.scss]: No longer applies.

**Review by delawski — APPROVED** [review]
Thanks for updating the code. Looks good to me now. I think it's ready to merge.

**Pull request merged**